django_mesh/views.py

```diff
@@ -135,4 +135,68 @@ class TagIndexView(ListView):
 
     def get_queryset(self, *args, **kwargs):
         qs = super(TagIndexView, self).get_queryset(*args, **kwargs)
-        return qs.get_for_user(self.request.user)
\ No newline at end of file
+        return qs.get_for_user(self.request.user)
+
+class MediaDetailView(DetailView):
+    model = Media
+    template_name = 'django_mesh/media_view.html'
+
+    def get_context_data(self, **kwargs):
+        context = super(MediaDetailView, self).get_context_data(**kwargs)
+        context['Media'] = self.model
+        return context
+
+    def get_queryset(self, *args, **kwargs):
+        qs = super(MediaDetailView, self).get_queryset(*args, **kwargs)
+        return qs.get_for_user(self.request.user)
+
+class MediaIndexView(ListView):
+    model = Media
+    template_name = 'django_mesh/media_index_view.html'
+
+    def get_queryset(self, *args, **kwargs):
+        qs = super(MediaIndexView, self).get_queryset(*args, **kwargs)
+        return qs.get_for_user(self.request.user)
+
+def self_enrollment(request, *args, **kwargs):
+    user = request.user
+    if request.method == 'POST':
+        channel = get_object_or_404(Channel.objects.get_for_user(user), slug=kwargs['slug'])
+        channel.followers.add(user)
+        return HttpResponseRedirect(reverse('mesh_channel_index'))
+    else:
+        return HttpResponseRedirect(reverse('mesh_channel_index'))
+
+class OembedDetailView(DetailView):
+
+    model = Media
+
+    @property
+    def get_site_domain(request):
+        return get_current_site(request).domain
+
+
+    def dispatch(self, request, *args, **kwargs):
+        self.media = get_object_or_404(Media.objects.get_for_user(user=self.request.user), slug=self.kwargs['slug'])
+
+        user = request.user
+
+        file_dict = {}
+
+        file_dict['version'] = 1.0
+        file_dict['title'] = self.media.title
+        file_dict['url'] = self.media.file_url
+        file_dict['provider_name'] = self.get_site_domain
+
+        if self.media.media_type == Media.MEDIA_TYPES.IMAGE:
+            file_dict['type'] = 'photo'
+            file_dict['width'] = 240
```

Comment:
Yes, good catch.

Yeah, let's just add nullable width and height SmallIntegerFields and use that instead of the hardcoded values.